packages/ncids-js/src/components/nci-header/utils/megamenu.ts

```diff
@@ -0,0 +1,557 @@
+		const selector = this.options.listElementClass;
+		return Array.from(this.element.querySelectorAll(<string>selector));
+	}
+
+	/**
+	 * Returns the HTMLElement that contains the MegaMenu
+	 *
+	 * @return {<HTMLElement>} MegaMenu DOM Element
+	 * @private
+	 */
+	private queryMegaMenu(button: Element): HTMLElement {
+		const selector = '.nci-header-megamenu';
+		const parent = <HTMLElement>button.parentElement;
+		return <HTMLElement>parent.querySelector(selector);
+	}
+
+	/**
+	 * Creates the primary navigation buttons from the link
+	 * element passed into it and adds aria tags
+	 *
+	 * @return {<HTMLElement>}  Button element with proper tags
+	 *
+	 * @param {Element} link anchor tag to be converted into button.
+	 * @private
+	 */
+	private activateButton(link: Element): Element {
+		const id = link.getAttribute('href') || '';
+		const button = this.createDom(
+			'button',
+			['nci-header-nav__primary-button'],
+			[{ 'aria-expanded': false }, { 'aria-controls': id }]
+		);
+		button.innerHTML = link.innerHTML;
+		link.replaceWith(button);
+		return link;
+	}
+
+	/**
+	 * Change the primary buttons back to links
+	 *
+	 * @param {Element} button anchor tag to be converted into button.
+	 * @return {<HTMLElement>}  Button element with proper tags
+	 * @private
+	 */
+	private deactivateButton(button: Element): Element {
+		const link = this.createDom('a', ['nci-header-nav__primary-link']);
+		link.innerHTML = button.innerHTML;
+		button.replaceWith(link);
+		return link;
+	}
+
+	/**
+	 * Finds newly updated buttons and adds event listeners.
+	 *
+	 * @private
+	 */
+	private findToggleButtons(): void {
+		const sections = this.queryListSections();
+		sections.forEach((section) => {
+			const button = section.firstElementChild;
+			this.toggleButtons.push(<HTMLElement>button);
+		});
+		this.addToggleEventListeners();
+	}
+
+	/**
+	 * Sets up event listeners for every possible button per collapsible.
+	 *
+	 * @private
+	 */
+	private addToggleEventListeners(): void {
+		[...this.toggleButtons].forEach((button) => {
+			button.addEventListener('click', this.eventListener);
+		});
+	}
+
+	/**
+	 * Removes event listeners for every possible button per collapsible.
+	 *
+	 * @private
+	 */
+	private removeToggleEventListeners(): void {
+		[...this.toggleButtons].forEach((button) => {
+			button.removeEventListener('click', this.eventListener);
+		});
+	}
+
+	/**
+	 * Click handler for the toggle buttons.
+	 *
+	 * @param {Event} event Collapsible section toggling event.
+	 *
+	 * @private
+	 */
+	private handleClick(event: Event): void {
+		event.preventDefault();
+		event.stopPropagation();
+		if (event.target instanceof HTMLElement) {
+			const button: HTMLElement = event.target;
+			this.toggleMegaMenu(button);
+		}
+	}
+
+	/**
+	 * Click handler for clicking outside of the menu
+	 *
+	 * @param {Event} event Event from document event handler.
+	 *
+	 * @private
+	 */
+	private handleClickOff(event: Event): void {
+		if (this.isActive) {
```

Comment:
This entire block is missing code coverage, but I'm not 100% sure if there are any quick fixes for the if on line 414 and 418? Line 415 might be able to get the same treatment with HTMLButtonElement from above